hicexplorer/hicPlotDistVsCounts.py

```diff
@@ -384,17 +385,13 @@ def main(args=None):
             axs[row, col] = ax
             idx += 1
             if args.outFileData is not None:
-                if args.perchr and len(args.matrices) > 1:
-                    label = labels[matrix_file]
-                    args.outFileData.write("#{}\n".format(chrom))
-
-                elif args.perchr:
-                    label = chrom
-                else:
-                    label = labels[matrix_file]
-                args.outFileData.write("#{}\n".format(label))
-                args.outFileData.write("\t".join(map(str, x)) + "\n")
-                args.outFileData.write("\t".join(map(str, y)) + "\n")
+                x_vals = np.stack(x).T
+                y_vals = np.stack(y).T
+                table_to_export = pd.DataFrame({'Matrix': matrix_file,
```

Comment:
can you try setting `labels[matrix_file]` instead of just `matrix_file`. Thus, instead of writing the table with the whole path to the file, only the label is saved.